plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -99,6 +111,28 @@ func UTF16ToUTF8Bytes(in []byte, length uint32) ([]byte, error) {
 	return decoded, err
 }
 
+func IsTheEndOfContent(in []byte, length uint32) bool {
+	// scan next 100 characters, if any of them is none '0', return false
+	i := int(length)
+
+	if i%2 != 0 {
+		i -= 1
+	}
+	max := len(in)
+	if i+100 < max {
```

Comment:
Is there any specific reason we try to scan 100 ahead? Or is this just arbitrary?

Reply:
Kind of arbitrary to avoid false negative caused by empty spaces, lines, bad luck with continuous '0's